completion/available/aliases.completion.bash

```diff
@@ -89,7 +93,7 @@ function _bash-it-component-completion-callback-on-init-aliases() {
                             prec_word=\${prec_word#* }
                         fi
                         (( COMP_CWORD += ${#alias_arg_words[@]} ))
-                        COMP_WORDS=(\"$alias_cmd\" \"${alias_arg_words[*]}\" \"\${COMP_WORDS[@]:1}\")
+                        COMP_WORDS=(\"$alias_cmd\" $(printf "'%s' " "${alias_arg_words[@]}") \"\${COMP_WORDS[@]:1}\")
```

Comment:
This still fails when `alias_arg_words[@]` contains `'`. You can use `%q `

Reply:
Did this in 15ec29d869b33a731828d3c653425145bc6a9151

Good call. Using `%q` does correctly escape things better.

I also had to switch to wrapping the `%q` output in double quotes rather than single quotes, which should be fine since everything is escaped now. If an alias contained single quotes, like the bash-it `reload_completion` alias does, this would produce invalid output. That is due to the complexity of escaping single quotes in single quotes and due to how that works, I believe it is impossible to make a single array entry that contains single quotes and is wrapped in single quotes.

Reply:
> I also had to switch to wrapping the `%q` output in double quotes rather than single quotes, which should be fine since everything is escaped now.

No, it doesn't work. `%q` already quotes everything needed. If you further quote it within `'...'` or `"..."`, the special characters for quoting is quoted and then remain in the final results.

```bash
$ a=('a b c')
$ eval "b=($(printf "\"%q\" " "${a[@]}"))"
$ declare -p a b
declare -a a=([0]="a b c")
declare -a b=([0]="a\\ b\\ c")
```

You should use `%q ` (but not `"%q" `).

Reply:
You are correct. My concern had been about a case like
```
myArray=("one two")
printf "%q " "${myArray[@]}"
```
That the print would produce `one two` when it needed to be `"one two"`. What `%q` actually does in a case like that is it escapes the space, which I didn't even think was possible, to produce `one\ two` which works just fine.

I removed the quotes in 3e9a03391d6bc3c72aadccf1f5a1a6a4861c4124, thanks for teaching me something new!